apps/server/src/modules/task/service/task.service.ts

```diff
@@ -23,6 +24,28 @@ export class TaskService {
 		return this.taskRepo.findBySingleParent(creatorId, courseId, filters, options);
 	}
 
+	async removeCreatorId(creatorId: EntityId): Promise<DomainOperation> {
```

Comment:
Please split it to two methods. It dosen't just remove creatorId

Reply:
@SevenWaysDP I agree that it would be better (and cleaner) to have this split across two separate methods, just wanted to agree on one thing: it should be split into two **private** methods (one for just the _creatorId_ removal and second one for the whole tasks deletion), but the exposed method (the public one) should stay being a single one - the reason behind that is that we want to have a single place to trigger a complete data deletion in any service, e.g., the _tasks_ service in this case, based on a given event (user deletion in this case). Although it's not really a hard requirement for now as we could easily handle two separate events with the results of both the data deletion and just the data modification, having a single public method for that is more future-proof as in the near future we'll probably want to change the implementation to use the event-based approach and to not do too many changes within the services it's better to keep just a single public method for triggering the data deletion for every case as in the event-based approach there would be just a single event, not many events based on some circumstances or anything.

@SevenWaysDP do you agree with that? So to sum up: still single public method (e.g. `removeCreatorIdOrDeleteTasks` and then two separate private methods (e.g. `removeCreatorId`, `deleteTasks`).

Reply:
@SevenWaysDP  @bn-pass I splitted this methdon into metod which delete task when task has no course or lesson, and second which update task without creator

Reply:
@SevenWaysDP we've discussed it and decided to go with your proposal - @WojciechGrancow will split it into separate public methods, it will be 3 in total as we'll also add put this userId deletion from the **archived** field into a separate method as it's basically another, separate operation. We've agreed that, even if we'll decide to switch to the event-based approach in the near future, we should be able to add some kind of a listener that could still implement a common API and just a single method (consume single event from the event bus) that could internally call all three service's methods.